resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java

```diff
@@ -309,6 +312,14 @@ public long queryTimeoutMillis() {
     }
 
     /**
+     * @param hostsFileEntriesResolver the {@link HostsFileEntriesResolver} used to first check
+     * if the hostname isn't locally aliased.
+     */
+    public void setHostsFileEntriesResolver(HostsFileEntriesResolver hostsFileEntriesResolver) {
```

Comment:
Actually, are all the mutable fields really necessary? Would one really mutate those?
Otherwise, it just seems to me that those are POJO style fields because the constructors is already overloaded so many times that it would be a real mess trying to support more final fields.
If that's the case, I strongly think we should have only one single constructor with all fields final, and move the construction logic/overloads in a builder pattern.